### ISSUE TITLE
Weather prints HTML file

### ISSUE THREAD

**Issue by Thunderbottom** [alexanderepstein/Bash-Snippets#10]
**Issue Label:**
* [x] bug
* [ ] feature
* [ ] enhancement

**Description of bug or feature enhancement:**
Typing in `weather` on macOS High Sierra returns an [HTML file](https://ptpb.pw/At1V/sh) instead of the expected output.

**OS:**
* [x] Mac
* [ ] Linux 32 Bit
* [ ] Linux 64 Bit
* [ ] Windows 32 Bit
* [ ] Windows 64 Bit

**System Specific Info:** 

```
OS version: macOS High Sierra Dev Preview 2
terminal: iTerm2 Nightly
shell: bash
```

**EDIT:** _Check [this](https://github.com/alexanderepstein/Bash-Snippets/issues/10#issuecomment-312094000) answer for the solution._

**Comment by alexanderepstein** [comment]
Whoah that seems unusual. I don't have access to a system with High Sierra installed on it but it has something to do with the curl call. @chubin do you know what is causing this?

**Comment by Thunderbottom** [comment]
Actually, do we even need to add `$location` to the curl call? Because it returns me results for my current location regardless of whether I put my location. 

Just calling `curl $locale.wttr.in` works fine as well. Although, it still shows the output as HTML.

**Comment by alexanderepstein** [comment]
It doesn't grab everybodys locations correctly (it doesn't work for me) that's why I query ip info and what not.

 Can you try using wget instead of curl and let me know what that returns.

**Comment by Thunderbottom** [comment]
Using `wget -qO- en.wttr.in` returns the same html page as before.

**Comment by chubin** [comment]
@Thunderbottom please do curl -vI wttr.in and show the result.

**Comment by Thunderbottom** [comment]
Shows the same HTML output.

**Comment by chubin** [comment]
It should show something like that:

```
$ curl -vI wttr.in
* Rebuilt URL to: wttr.in/
*   Trying 5.9.243.177...
* TCP_NODELAY set
* Connected to wttr.in (5.9.243.177) port 80 (#0)
> HEAD / HTTP/1.1
> Host: wttr.in
> User-Agent: curl/7.52.1
> Accept: */*
> 
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< Server: nginx/1.9.14
Server: nginx/1.9.14
< Date: Thu, 29 Jun 2017 18:05:20 GMT
Date: Thu, 29 Jun 2017 18:05:20 GMT
< Content-Type: text/html; charset=utf-8
Content-Type: text/html; charset=utf-8
< Content-Length: 9336
Content-Length: 9336
< Connection: keep-alive
Connection: keep-alive

< 
* Curl_http_done: called premature == 0
* Connection #0 to host wttr.in left intact

```

Are you sure that you are using curl?
What to you get if you do:

```
curl --version
```

?

**Comment by Thunderbottom** [comment]
Alright, I mistook `curl -vI` for `curl -vl`

Here is the output : 

```shell
$ curl -vI wttr.in
* Rebuilt URL to: wttr.in/
*   Trying 5.9.243.177...
* TCP_NODELAY set
* Connected to wttr.in (5.9.243.177) port 80 (#0)
> HEAD / HTTP/1.1
> Host: wttr.in
> User-Agent: Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)
> Accept: */*
> Referer:
>
< HTTP/1.1 200 OK
HTTP/1.1 200 OK
< Server: nginx/1.9.14
Server: nginx/1.9.14
< Date: Thu, 29 Jun 2017 19:16:55 GMT
Date: Thu, 29 Jun 2017 19:16:55 GMT
< Content-Type: text/html; charset=utf-8
Content-Type: text/html; charset=utf-8
< Content-Length: 33259
Content-Length: 33259
< Connection: keep-alive
Connection: keep-alive

<
* Connection #0 to host wttr.in left intact
```

> Are you sure that you are using curl?

Of course.

> curl --version

```
$ curl --version
curl 7.54.0 (x86_64-apple-darwin17.0) libcurl/7.54.0 LibreSSL/2.0.20 zlib/1.2.8 nghttp2/1.20.0
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: AsynchDNS IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz HTTP2 UnixSockets HTTPS-proxy
```

**Comment by Thunderbottom** [comment]
Apparently the User Agent is different for both of us

Yours:

```
> User-Agent: curl/7.52.1
```

Mine:

```
> User-Agent: Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0)
```

**Comment by chubin** [comment]
I don't known why your curl uses this user-agent string, it's obviously wrong (probably you've configured it in ~/.curlrc), but you can override it with `-A`:

```
curl -Acurl wttr.in
```

@alexanderepstein You can force user agent in the weather script using -A. It will guarantee that failures like that will not happen (and the method hat no drawbacks at all, at least not for the clients).

**Comment by alexanderepstein** [comment]
@chubin 
Just going to get confirmation that this solves the issue (although after looking into it I think it will) and will implement it in the next commit

**Comment by Thunderbottom** [comment]
Yes, using the `-A` flag makes it work right. @alexanderepstein you should implement it in the next commit.

**Comment by alexanderepstein** [comment]
Added in commit f7aaed704c15a3e4db42d9749c5e59ac32734474

**Comment by alexanderepstein** [comment]
Released in version 1.2.1
